cleanlab/multiannotator.py

```diff
@@ -0,0 +1,859 @@
+
+        # tiebreak 2: using empirical class frequencies
+        if len(tied_idx) > 0:
+            class_frequencies = (
+                labels_multiannotator.apply(lambda s: s.value_counts(), axis=1).sum().values
+            )
+            for idx, label_mode in tied_idx.copy().items():
+                max_frequency = np.where(
+                    class_frequencies[label_mode] == class_frequencies[label_mode].max()
+                )[0]
+                if len(max_frequency) == 1:
+                    consensus_label[idx] = label_mode[max_frequency[0]]
+                    del tied_idx[idx]
+                else:
+                    tied_idx[idx] = label_mode[max_frequency]
+
+        # tiebreak 3: using initial annotator quality scores
+        if len(tied_idx) > 0:
+            nontied_consensus_label = consensus_label[nontied_idx]
+            nontied_labels_multiannotator = labels_multiannotator.iloc[nontied_idx]
+            annotator_agreement_with_consensus = nontied_labels_multiannotator.apply(
+                lambda s: np.mean(s[pd.notna(s)] == nontied_consensus_label[pd.notna(s)]),
+                axis=0,
+            ).to_numpy()
+            for idx, label_mode in tied_idx.copy().items():
+                label_quality_score = np.array(
+                    [
+                        np.mean(
+                            annotator_agreement_with_consensus[
+                                labels_multiannotator.iloc[idx][
+                                    labels_multiannotator.iloc[idx] == label
+                                ].index.values
+                            ]
+                        )
+                        for label in label_mode
+                    ]
+                )
+                max_score = np.where(label_quality_score == label_quality_score.max())[0]
+                if len(max_score) == 1:
+                    consensus_label[idx] = label_mode[max_score[0]]
+                    del tied_idx[idx]
+                else:
+                    tied_idx[idx] = label_mode[max_score]
+
+        # if still tied, break by random selection
+        if len(tied_idx) > 0:
+            warnings.warn(
+                f"breaking ties of examples {list(tied_idx.keys())} by random selection, you may want to set seed for reproducability"
+            )
+            for idx, label_mode in tied_idx.items():
+                consensus_label[idx] = np.random.choice(label_mode)
+
+    else:
+        raise ValueError(
+            f"""
+            {consensus_method} is not a valid consensus method!
+            Please choose a valid consensus_method: {valid_methods}
+            """
+        )
+
+    return consensus_label.astype("int64")
+
+
+def _get_annotator_agreement(
+    labels_multiannotator: pd.DataFrame,
+    consensus_label: np.ndarray,
+) -> np.ndarray:
+    """Returns the fractions of annotators that agree with the consensus labels. Note that the
```

Comment:
[nit] Returns the fraction of annotators that agree with the consensus label per example.